spec/jobs/renewal_job_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe RenewCheckoutJob, type: :job do
+  describe '#perform_later' do
+    let(:mock_sc_client) { instance_double(SymphonyClient) }
+    let(:ws_args) { { item_key: 1,
+                      resource: '/catalog/item',
+                      session_token: '1s2fa21465' } }
+    let(:renewal_response) { RENEWAL_RAW_JSON.to_json }
```

Comment:
Have a suggestion for eliminating a duplicative stubbed data: just use sinatra here too like I did on view holds:

```rb
# frozen_string_literal: true

require 'rails_helper'

RSpec.describe RenewCheckoutJob, type: :job do
  describe '#perform_later' do
    let(:ws_args) { { item_key: 1,
                      resource: '/catalog/item',
                      session_token: '1s2fa21465' } }

    before do
      stub_request(:any, /example.com/).to_rack(FakeSymphony)
    end

    after do
      Redis.current.flushall
    end

    context 'with valid input that is returned OK from SymphonyClient' do
      it 'sets a Redis value that marks success' do
        described_class.perform_now(**ws_args)
        results = Redis.current.get 'renewal_1'

        expect(results).to include 'success'
      end

      it 'sets a Redis value that contains a translated item key' do
        described_class.perform_now(**ws_args)
        results = Redis.current.get 'renewal_1'
        renewal_period = 85 # days
        new_due_date_formatted =  (Date.today + renewal_period.days).strftime('%B %-d, %Y')

        expect(results).to eq "{\"item_key\":1,\"result\":\"success\",\"renewal_count\":70,"\
                                      "\"due_date\":\"#{new_due_date_formatted}\",\"status\":null}"
      end
    end
```
And would have to stub an additional bad request to example.com for the failure...

Reply:
yes, great idea! thanks, I will make the change

Reply:
this is also updated with the last commit